### PR TITLE
nfs.py: handling host key verification failure

### PR DESCRIPTION
**Pull request by balamuruhans**
Configuring nfs remotely might fail if host key expires,
this patch handles and updates it that avoids offending
of host

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
**Before Fix**
```
2017-01-27 05:18:07,884 nfs              L0345 DEBUG| Mount 9.47.70.201:/var/lib/libvirt/images/NFS to /var/lib/libvirt/images/sharing
2017-01-27 05:18:07,885 process          L0368 INFO | Running 'ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw''
2017-01-27 05:18:07,928 process          L0462 DEBUG| [stderr] @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
2017-01-27 05:18:07,929 process          L0482 INFO | Command 'ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw'' finished with 255 after 0.0398750305176s
2017-01-27 05:18:07,929 process          L0462 DEBUG| [stderr] @    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
2017-01-27 05:18:07,929 process          L0462 DEBUG| [stderr] @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
2017-01-27 05:18:07,930 process          L0462 DEBUG| [stderr] IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
2017-01-27 05:18:07,930 process          L0462 DEBUG| [stderr] Someone could be eavesdropping on you right now (man-in-the-middle attack)!
2017-01-27 05:18:07,930 process          L0462 DEBUG| [stderr] It is also possible that a host key has just been changed.
2017-01-27 05:18:07,930 process          L0462 DEBUG| [stderr] The fingerprint for the ECDSA key sent by the remote host is
2017-01-27 05:18:07,930 process          L0462 DEBUG| [stderr] SHA256:GgpqGhHyxWvOqTAgUyPKONbQL/sT2nM1eeNriKAYtIw.
2017-01-27 05:18:07,931 process          L0462 DEBUG| [stderr] Please contact your system administrator.
2017-01-27 05:18:07,931 process          L0462 DEBUG| [stderr] Add correct host key in /root/.ssh/known_hosts to get rid of this message.
2017-01-27 05:18:07,931 process          L0462 DEBUG| [stderr] Offending ECDSA key in /root/.ssh/known_hosts:1
2017-01-27 05:18:07,931 process          L0462 DEBUG| [stderr]   remove with:
2017-01-27 05:18:07,931 process          L0462 DEBUG| [stderr]   ssh-keygen -f "/root/.ssh/known_hosts" -R 9.47.70.203
2017-01-27 05:18:07,932 process          L0462 DEBUG| [stderr] ECDSA host key for 9.47.70.203 has changed and you have requested strict checking.
2017-01-27 05:18:07,932 process          L0462 DEBUG| [stderr] Host key verification failed.
2017-01-27 05:18:07,932 virsh_migrate    L1127 ERROR| <class 'avocado.core.exceptions.TestFail'>: Failed to run: ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw'
```

**After Fix:**

```
2017-01-27 06:38:54,654 nfs              L0345 DEBUG| Mount 9.47.70.201:/var/lib/libvirt/images/NFS to /var/lib/libvirt/images/sharing
2017-01-27 06:38:54,655 process          L0368 INFO | Running 'ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw''
2017-01-27 06:38:54,698 process          L0482 INFO | Command 'ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw'' finished with 255 after 0.0391728878021s
2017-01-27 06:38:54,698 process          L0462 DEBUG| [stderr] @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
2017-01-27 06:38:54,699 process          L0462 DEBUG| [stderr] @    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
2017-01-27 06:38:54,699 process          L0462 DEBUG| [stderr] @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
2017-01-27 06:38:54,699 process          L0462 DEBUG| [stderr] IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
2017-01-27 06:38:54,699 process          L0462 DEBUG| [stderr] Someone could be eavesdropping on you right now (man-in-the-middle attack)!
2017-01-27 06:38:54,699 process          L0462 DEBUG| [stderr] It is also possible that a host key has just been changed.
2017-01-27 06:38:54,700 process          L0462 DEBUG| [stderr] The fingerprint for the ECDSA key sent by the remote host is
2017-01-27 06:38:54,700 process          L0462 DEBUG| [stderr] SHA256:GgpqGhHyxWvOqTAgUyPKONbQL/sT2nM1eeNriKAYtIw.
2017-01-27 06:38:54,700 process          L0462 DEBUG| [stderr] Please contact your system administrator.
2017-01-27 06:38:54,700 process          L0462 DEBUG| [stderr] Add correct host key in /root/.ssh/known_hosts to get rid of this message.
2017-01-27 06:38:54,700 process          L0462 DEBUG| [stderr] Offending ECDSA key in /root/.ssh/known_hosts:1
2017-01-27 06:38:54,701 process          L0462 DEBUG| [stderr]   remove with:
2017-01-27 06:38:54,701 process          L0462 DEBUG| [stderr]   ssh-keygen -f "/root/.ssh/known_hosts" -R 9.47.70.203
2017-01-27 06:38:54,701 process          L0462 DEBUG| [stderr] ECDSA host key for 9.47.70.203 has changed and you have requested strict checking.
2017-01-27 06:38:54,701 process          L0462 DEBUG| [stderr] Host key verification failed.
2017-01-27 06:38:54,702 process          L0368 INFO | Running 'ssh-keygen -R 9.47.70.203'
2017-01-27 06:38:58,096 process          L0462 DEBUG| [stdout] # Host 9.47.70.203 found: line 1
2017-01-27 06:38:58,096 process          L0482 INFO | Command 'ssh-keygen -R 9.47.70.203' finished with 0 after 0.143362998962s
2017-01-27 06:38:58,097 process          L0462 DEBUG| [stdout] /root/.ssh/known_hosts updated.
2017-01-27 06:38:58,097 process          L0462 DEBUG| [stdout] Original contents retained as /root/.ssh/known_hosts.old
2017-01-27 06:39:05,268 process          L0368 INFO | Running 'ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw''
2017-01-27 06:39:05,268 process          L0482 INFO | Command 'ssh root@9.47.70.203  'mount -t nfs 9.47.70.201:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw'' finished with 0 after 7.17168688774s
```

**Comment by balamuruhans** [comment]
```
2017-04-05 09:20:59,271 process          L0368 INFO | Running 'ssh-keygen -R 9.40.192.88'
2017-04-05 09:20:59,301 process          L0462 DEBUG| [stdout] # Host 9.40.192.88 found: line 26 type ECDSA
2017-04-05 09:20:59,302 process          L0482 INFO | Command 'ssh-keygen -R 9.40.192.88' finished with 0 after 0.0251858234406s
2017-04-05 09:20:59,302 process          L0462 DEBUG| [stderr] /root/.ssh/known_hosts updated.
2017-04-05 09:20:59,303 process          L0462 DEBUG| [stderr] Original contents retained as /root/.ssh/known_hosts.old
2017-04-05 09:20:59,305 remote           L0362 DEBUG| Attempting to log into 9.47.68.169:22 using ssh (timeout 240s)
2017-04-05 09:21:07,010 client           L1049 DEBUG| Sending command: /bin/ssh-agent /bin/sh
2017-04-05 09:21:07,140 client           L1049 DEBUG| Sending command: echo $?
2017-04-05 09:21:07,241 client           L1049 DEBUG| Sending command: /bin/ssh-add /root/.ssh/id_rsa
2017-04-05 09:21:07,350 client           L1049 DEBUG| Sending command: echo $?
2017-04-05 09:21:17,946 utils_conn       L0649 DEBUG| SSH connection setup successfully.
2017-04-05 09:21:17,946 nfs              L0279 DEBUG| The command: ssh root@9.40.192.88 'mount | grep -E '.*/var/lib/libvirt/images/NFS.*/var/lib/libvirt/images/sharing.*''
2017-04-05 09:21:24,169 nfs              L0282 DEBUG| The command result: <256:>
2017-04-05 09:21:24,170 nfs              L0344 DEBUG| To check if the /var/lib/libvirt/images/sharing exists
2017-04-05 09:21:30,444 nfs              L0348 DEBUG| Prepare to create /var/lib/libvirt/images/sharing
2017-04-05 09:21:36,626 nfs              L0356 DEBUG| Mount 9.47.68.169:/var/lib/libvirt/images/NFS to /var/lib/libvirt/images/sharing
2017-04-05 09:21:36,627 process          L0368 INFO | Running 'ssh root@9.40.192.88 -o StrictHostKeyChecking=no  'mount -t nfs 9.47.68.169:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw''
2017-04-05 09:21:45,057 process          L0482 INFO | Command 'ssh root@9.40.192.88 -o StrictHostKeyChecking=no  'mount -t nfs 9.47.68.169:/var/lib/libvirt/images/NFS /var/lib/libvirt/images/sharing -o rw'' finished with 0 after 8.42495417595s
```

**Comment by xutian** [comment]
@balamuruhans  fixed pep8 e722 issue, can you refresh your branch to re-trigger CI checking.

Thanks,
Xu

**Comment by balamuruhans** [comment]
@xutian sure.

File: virttest/nfs.py [hunk 1]
```diff
@@ -254,10 +260,13 @@ def __init__(self, params):
...
         self.mount_options = params.get("nfs_mount_options")
         self.mount_src = params.get("nfs_mount_src")
-        self.nfs_client_ip = params.get("nfs_client_ip")
         self.nfs_server_ip = params.get("nfs_server_ip")
         self.ssh_user = params.get("ssh_username", "root")
         self.remote_nfs_mount = params.get("remote_nfs_mount", "yes")
+        self.ssh_hostkey_check = params.get("ssh_hostkey_check", "no") == "yes"
+        self.ssh_cmd = "ssh %s@%s " % (self.ssh_user, self.nfs_client_ip)
+        if not self.ssh_hostkey_check:
+            self.ssh_cmd += "-o StrictHostKeyChecking=no "
```
Comment (will-Do) [inline virttest/nfs.py]: :+1:

**Comment by balamuruhans** [comment]
Thanks @will-Do @xutian

**Pull request merged**